src/codegen/generators/expression_generator.rs

```diff
@@ -657,14 +672,32 @@ impl<'ink, 'b> ExpressionCodeGenerator<'ink, 'b> {
 
             if let Some((declaration_type, type_name)) = param {
                 let argument: BasicValueEnum = if declaration_type.is_by_ref() {
-                    self.generate_argument_by_ref(param_statement, type_name)?
+                    let declared_parameter = declared_parameters.get(location);
+                    self.generate_argument_by_ref(param_statement, type_name, declared_parameter)?
                 } else {
                     //pass by val
                     self.generate_argument_by_val(type_name, param_statement)?
                 };
                 result.push((location, argument));
             }
+
+            passed_args.push(location);
+        }
+
+        // handle missing declared parameters
+        if declared_parameters.len() > passed_args.len() {
```

Comment:
if you use : 
`for param in declared_parameters.iter().skip(argument.len())` you do not need the passed_args variable.